lib/puppet/provider/augeas/augeas.rb

```diff
@@ -203,7 +203,7 @@ def close_augeas
 
   def is_numeric?(s)
     case s
-    when Fixnum
+    when Integer,Float
```

Comment:
While I agree that it looks like a bug in the original, this is not a change that is required to make puppet run on Ruby 2.4.0. After this change, more values are allowed than earlier (floats in float form).